sources/api/thar-be-settings/src/service.rs

```diff
@@ -4,36 +4,49 @@ use std::collections::HashSet;
 use std::path::Path;
 use std::process;
 
-use itertools::join;
+use itertools::{join, Itertools};
 
 use crate::{error, Result};
 
-/// Wrapper for the multiple functions needed to go from
-/// a list of changed settings to a Services map
-#[allow(clippy::implicit_hasher)]
-pub async fn get_affected_services<P>(
+//returns changed settings corresponding to the services
+pub async fn get_affected_services_settings_map<P>(
     socket_path: P,
     settings_limit: Option<HashSet<String>>,
-) -> Result<model::Services>
+) -> Result<Option<HashMap<String, Vec<String>>>>
 where
     P: AsRef<Path>,
 {
-    let service_limit = if let Some(settings_limit) = settings_limit {
+    let settings_limit = if let Some(settings_limit) = settings_limit {
         let setting_to_service_map =
             get_affected_service_map(socket_path.as_ref(), settings_limit).await?;
```

Comment:
I would refractor function name and some variable names here, since it seems similar and confusing